### PR TITLE
Search Bar

### LINKED ISSUE #809: Add search bar in simulator
As a new user of CircuitVerse, I don't know in which categories all components are so it will be helpful to add a small search bar

**Additional context**
image
something like this but make it look as a search bar
image

_This issue is a part of my [Usability Test](https://drive.google.com/open?id=1igwLx2tc1dAU_75qY2F4u7ZnHf2crcp-)_
**Are you working on this? (Yes/No)**
No

### PR DESCRIPTION
**Pull request by NumberC**
Fixes #809 

#### Describe the changes you have made in this pr -
Added search bar in simulator

### Screenshots of the changes (If any) -
Search
Normal

### CODE REVIEW

**Comment by shubhankarsharma00** [comment]
please squash the commits

**Comment by NumberC** [comment]
@shubhankarsharma00 squash the commit up to "Basis of search"? Or were you referring to another group of commits?

**Comment by Ananthavijay** [comment]
@NumberC Rebase the branch, remove the commits which are not related to this PR and squash the commits.

File: app/views/simulator/edit.html.erb [hunk 1]
```diff
@@ -77,7 +77,7 @@
     <div class="modules noSelect defaultCursor" >
       <div id="modules-header">Circuit Elements</div>
       <div id="searchBar">
-        <input type="text" oninput="searchResults(this)">
+        <input type="text" style="width: 100%; margin-bottom: 10px;" placeholder="Search Here" oninput="searchResults(this)">
```
[the remark this answers is not in the document]
Comment (NumberC) [reply app/views/simulator/edit.html.erb]: Yes, yes it would. Thank you!

**Comment by NumberC** [comment]
@shubhankarsharma00 Can you please help? I tried to squash all the old commits and this happened.

**Comment by shubhankarsharma00** [comment]
@NumberC the easiest way would be to use git reset soft. Reset your history to upstream master's head. Then a git commit will do it.

**Comment by NumberC** [comment]
@shubhankarsharma00 Thank you, but won't that also delete all my commits in this branch?

**Comment by shubhankarsharma00** [comment]
Reset soft will delete your commit history but not the files. One git commit will give you a single commit for all the changes made.

**Comment by NumberC** [comment]
@shubhankarsharma00 I did "git reset --soft upstream/master" and this happened. I don't know how to get any of my work back.

**Comment by NumberC** [comment]
@shubhankarsharma00 Everything is here now https://github.com/CircuitVerse/CircuitVerse/pull/838

**Comment by NumberC** [comment]
@shubhankarsharma00 Nevermind. It's here now: https://github.com/CircuitVerse/CircuitVerse/pull/839

**Pull request closed without merging**